aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -188,12 +188,35 @@ static TensorIterator make_reduction(
 
 Tensor _cumsum_cpu(const Tensor& self, int64_t dim) {
   Tensor result = at::empty_like(self, MemoryFormat::Contiguous);
-  cumsum_stub(self.device().type(), result, self, dim);
+  _cumsum_out_cpu(result, self, dim);
   return result;
 }
 
 Tensor& _cumsum_out_cpu(Tensor& result, const Tensor& self, int64_t dim) {
-  cumsum_stub(self.device().type(), result, self, dim);
+  if (result.numel() == 0) {
+    result.resize_(self.sizes());
+  }
+  if (self.dim() == 0) {
+    result.fill_(self);
+    return result;
+  }
+  if (self.numel() == 0) {
+    return result;
+  }
+
+  TensorIterator iter = dim_apply_op(result, self, maybe_wrap_dim(dim, self.dim()));
+  AT_DISPATCH_ALL_TYPES(iter.dtype(), "cumsum_out_cpu", [&]{
```

Comment:
You should try to move it in kernel code (to compile with AVX) it might give speedup.

Reply:
No benefit to gain with dim apply case, discussed with Vitaly, we decide to keep this out of cpu folder.